### PR TITLE
Ops 3808 planned maintenance metrics

### PR DESCRIPTION
**Pull request by M-Schiborr**
# Description
Planned Maintenance Module for dbcmetrics

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by YannickEvers — APPROVED** [review]

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                        platform_window_start = self.parse_string_to_datetime(maintance_entry['scheduled_at'])
+                    except Exception as e:
+                        # Triggered if field for start of platform maintenance window is empty
+                        logging.warning(f"Couldn't load maintenance entry: platform_window_start on {platform_name} for entry: {maintance_entry}")
+                        logging.exception(f"\t {e}")
+                        continue
+
+                    try:
+                        message_string: str = maintance_entry['message']
+                        stunde_result = re.findall(r"([\d]+)\s*h", message_string) # returns e.g.: 2, 10, 1 as String in a List
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcplannedmaintenance.py:87]: The regex should be more specific. For example it also matches hier10 hallo

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                        start_to_end_window_hours: int = int(stunde_result[0])
+                        succsessfully_parsed_hours: bool = True
+                    except Exception as e:
+                        logging.warning(f"\tCouldn't parse hours of end of maintenance window, using default hour offset (0 hours).")
+                        succsessfully_parsed_hours: bool = False
+                        start_to_end_window_hours: int = 0
+                    
+                    try:
+                        minute_result = re.findall(r'([\d]+)\s*m', message_string) # returns e.g.: 2, 10, 40 as String in a List
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcplannedmaintenance.py:96]: Here the same

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                        start_to_end_window_min: int = int(minute_result[0])
+                    except Exception as e:
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcplannedmaintenance.py:98]: Do I see it right that, when only hours or minutes are defined, a warning about not parsable maintenance window is produced?

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                        logging.warning(f"\tCouldn't parse minutes of end of maintenance window, using default offset ({self.PLANNNED_MAINTENANCE_DEFAULT_DURATION}min) if parsining hours failed too.")
+                        if(succsessfully_parsed_hours):
+                            # if hours are correctly parse -> dont add default minutes 
+                            start_to_end_window_min: int = 0
+                        else:
+                            start_to_end_window_min: int = self.PLANNNED_MAINTENANCE_DEFAULT_DURATION
+
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcplannedmaintenance.py:105]: Please remove not needed whitespaces

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            if in_window:
+                self.metric.labels(platform=platform_name).set(1)
+            else:
+                self.metric.labels(platform=platform_name).set(0)
+
+
+    @staticmethod
+    def currently_in_window(platform_window_start: datetime.datetime, platform_window_end: datetime.datetime) -> bool:
+
+        tz_berlin = pytz.timezone('Europe/Berlin')
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcplannedmaintenance.py:140]: It's not a good practice to hardcode the used timezone

**Review by dimapin — CHANGES_REQUESTED** [review]

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py (hunk 1 above)
Comment (M-Schiborr) [reply dbcmetrics/dbcm_logic/dbcplannedmaintenance.py]: > The regex should be more specific. For example it also matches hier10 hallo

Has now been improved

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py (hunk 2 above)
Comment (M-Schiborr) [reply dbcmetrics/dbcm_logic/dbcplannedmaintenance.py]: Has now been improved

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py (hunk 4 above)
Comment (M-Schiborr) [reply dbcmetrics/dbcm_logic/dbcplannedmaintenance.py]: Has now been improved

File: dbcmetrics/dbcm_logic/dbcplannedmaintenance.py (hunk 5 above)
Comment (M-Schiborr) [reply dbcmetrics/dbcm_logic/dbcplannedmaintenance.py]: Has now been improved

**Review by dimapin — APPROVED** [review]

**Comment by M-Schiborr** [comment]
Merge OPS-3808 Planned Maintenance Metrics

**Pull request merged**